src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```diff
@@ -283,6 +284,99 @@ public class LockFileBuilder
             return lockFile;
         }
 
+        private Dictionary<Tuple<string, NuGetVersion>, LockFileLibrary> GetUniqueLockFileLibraries(LockFile lockFile)
```

Comment:
This method is called `GetUniqueLockFileLibraries`, while the other one is `EnsureUniqueLockFileTargetLibraries`, but they really are doing the same thing. 

I'd recommend making the naming pattern match.